std/traits.d

```diff
@@ -2279,15 +2279,18 @@ usually used for compile-time reflection purposes.
 template FunctionTypeOf(func...)
 if (func.length == 1 && isCallable!func)
 {
-    static if (is(typeof(& func[0]) Fsym : Fsym*) && is(Fsym == function) || is(typeof(& func[0]) Fsym == delegate))
+    static if ((is(typeof(& func[0]) Fsym : Fsym*) && is(Fsym == function)) || is(typeof(& func[0]) Fsym == delegate))
     {
         alias FunctionTypeOf = Fsym; // HIT: (nested) function symbol
     }
-    else static if (is(typeof(& func[0].opCall) Fobj == delegate))
+    else static if (is(typeof(& func[0].opCall) Fobj == delegate) || is(typeof(& func[0].opCall!()) Fobj == delegate))
```

Comment:
Can someone provide me with an example of where this is applicable?

Reply:
```d
void fun()
{
    int a;
    struct B { int opCall() { return a; }}
}
```
}

Reply:
Done